src/Riok.Mapperly/Descriptors/MappingBodyBuilders/BuilderContext/MembersMappingBuilderContext.cs

```diff
@@ -201,3 +216,44 @@ private void AddMappedAndIgnoredSourceMembersDiagnostics()
         }
     }
 }
+
+public static class TypeExtensions
+{
+    public static bool IsComplexType(this ITypeSymbol type, WellKnownTypes knowTypes)
+    {
+        return type.TypeKind switch
+        {
+            // check if it's an IEnumerable of primitives
+            TypeKind.Class
+                => type.Name != "string"
+                    && (
+                        (
+                            type is INamedTypeSymbol namedType
+                            && namedType.AllInterfaces.Any(s => string.Equals(s.Name, "IEnumerable", StringComparison.OrdinalIgnoreCase))
+                            && namedType.TypeArguments.Count() > 0
+                            && namedType.TypeArguments[0].IsComplexType(knowTypes)
+                        )
+                        || !(
+                            type is INamedTypeSymbol namedType2
+                            && namedType2.AllInterfaces.Any(s => string.Equals(s.Name, "IEnumerable", StringComparison.OrdinalIgnoreCase))
+                        )
+                    ),
+            TypeKind.Struct => false,
```

Comment:
It looks like this would accept any struct type, not just primitives. Is this intentional?

Reply:
I actually did this, because in C# some system types are structs (decimal, DateOnly, etc.). However, I've rewritten this logic to be clearer.